src/iop/highlights.c

```diff
@@ -2112,10 +2156,11 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
     case DT_IOP_HIGHLIGHTS_SEGMENTS:
     {
       const dt_segments_mask_t vmode = ((g != NULL) && fullpipe) ? g->segmentation_mask_mode : DT_SEGMENTS_MASK_OFF;
-      const gboolean anyclip = _process_opposed(piece, ivoid, ovoid, roi_in, roi_out, data);
-      const gboolean complete = fullpipe || (piece->pipe->type & DT_DEV_PIXELPIPE_EXPORT);
-      if(complete && (anyclip || (vmode != DT_SEGMENTS_MASK_OFF)))
-        _process_segmentation(piece, ivoid, ovoid, roi_in, roi_out, data, vmode);
+
+      float *tmp = _process_opposed(piece, ivoid, ovoid, roi_in, roi_out, data);
+      if(tmp)
+        _process_segmentation(piece, ivoid, ovoid, roi_in, roi_out, data, vmode, tmp);
```

Comment:
There is a check if tmp is true (not null) to do _process_segmentation.
But there is a blind dt_free_align() call?

So what if float *tmp = null. Then no _process_segmentation() is called, but dt_free_align(null) is called. Is this safe?

Reply:
Yes. Safe and as intended.